File: 20220727/모범답안/20220727_09.js/CoronaKeepDistance.js
/**
 * @param lineUp {number[]}
 * @param level {number}
 * @returns {boolean}
 */
function solution(lineUp, level) {
  let distanceSoFar = 9999
  for (let num of lineUp) {
    if (num === 1) {
      if (distanceSoFar < level) return false
      distanceSoFar = 0
    } else if (num === 0) {
      distanceSoFar++
    }
  }
  return true
}

solution

File: 20220727/모범답안/20220727_09.js/CoronaKeepDistance.py
def solution(lineUp, level):
    """
    :param lineUp: int[]
    :param level: int
    :return: boolean
    """

    distanceSoFar = 9999
    for num in lineUp:
        if num == 1:
            if distanceSoFar < level: return False
            distanceSoFar = 0
        elif num == 0:
            distanceSoFar += 1

    return True
